src/Microsoft.DotNet.Tools.Resgen/ResourceAssemblyGenerator.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.Tools.Resgen
+{
+    internal class ResourceAssemblyGenerator
+    {
+        public static void Generate(ResourceSource[] sourceFiles, Stream outputStream, string asemblyName, string culture, string[] references)
+        {
+            if (sourceFiles == null)
+            {
+                throw new ArgumentNullException(nameof(sourceFiles));
+            }
+
+            if (outputStream == null)
+            {
+                throw new ArgumentNullException(nameof(outputStream));
+            }
+
+            if (!sourceFiles.Any())
+            {
+                throw new InvalidOperationException("No source files specified");
+            }
+
+            var resourceDescriptions = new List<ResourceDescription>();
+            foreach (var resourceInputFile in sourceFiles)
+            {
+                if (resourceInputFile.Resource.Type == ResourceFileType.Resx)
+                {
+                    resourceDescriptions.Add(new ResourceDescription(
+                        resourceInputFile.MetadataName,
+                        () => GenerateResources(resourceInputFile.Resource), true));
+
+                }
+                else if (resourceInputFile.Resource.Type == ResourceFileType.Resources)
+                {
+                    resourceDescriptions.Add(new ResourceDescription(resourceInputFile.Resource.File.Name, () => resourceInputFile.Resource.File.OpenRead(), true));
+                }
+                else
+                {
+                    throw new InvalidOperationException("Generation of resource assemblies from dll not supported");
+                }
+            }
+
+            var compilationOptions = new CSharpCompilationOptions(outputKind: OutputKind.DynamicallyLinkedLibrary);
+            var compilation = CSharpCompilation.Create(asemblyName,
+                references: references.Select(reference => MetadataReference.CreateFromFile(reference)),
+                options: compilationOptions);
+
+            if (!string.IsNullOrEmpty(culture))
+            {
+                compilation = compilation.AddSyntaxTrees(new[]
```

Comment:
Are we going to need to worry about making sure the assembly version of the resources dll matches the neutral assembly? Anyone know?